File: ClassEval_T/cpp/solution/BookManagement.cpp

#include <string>
#include <unordered_map>
#include <stdexcept>

class BookManagement {
public:
    BookManagement();

    void add_book(const std::string& title, int quantity = 1);
    void remove_book(const std::string& title, int quantity);
    std::unordered_map<std::string, int> view_inventory() const;
    int view_book_quantity(const std::string& title) const;

private:
    std::unordered_map<std::string, int> inventory;
};

BookManagement::BookManagement() {}

void BookManagement::add_book(const std::string& title, int quantity) {
    if (inventory.find(title) != inventory.end()) {
        inventory[title] += quantity;
    }
    else {
        inventory[title] = quantity;
    }
}

void BookManagement::remove_book(const std::string& title, int quantity) {
    auto it = inventory.find(title);
    if (it == inventory.end() || it->second < quantity) {
        throw std::runtime_error("Insufficient quantity or book not found");
    }
    it->second -= quantity;
    if (it->second == 0) {
        inventory.erase(it);
    }
}

std::unordered_map<std::string, int> BookManagement::view_inventory() const {
    return inventory;
}

int BookManagement::view_book_quantity(const std::string& title) const {
    auto it = inventory.find(title);
    if (it == inventory.end()) {
        return 0;
    }
    return it->second;
}

File: BatchTestTool/java/java automated tester/Test/compile_passed/solutions/BookManagement.java
import java.util.HashMap;
import java.util.Map;

public class BookManagement {
    private Map<String, Integer> inventory;

    public BookManagement() {
        inventory = new HashMap<>();
    }

    public void addBook(String title, int quantity) {
        if (inventory.containsKey(title)) {
            inventory.put(title, inventory.get(title) + quantity);
        } else {
            inventory.put(title, quantity);
        }
    }

    public void removeBook(String title, int quantity) throws Exception {
        if (!inventory.containsKey(title) || inventory.get(title) < quantity) {
            throw new Exception("Book not found or insufficient quantity");
        }
        inventory.put(title, inventory.get(title) - quantity);
        if (inventory.get(title) == 0) {
            inventory.remove(title);
        }
    }

    public int viewBookQuantity(String title) {
        if (!inventory.containsKey(title)) {
            return 0;
        }
        return inventory.get(title);
    }

    public Map<String, Integer> viewInventory() {
        return inventory;
    }
}